File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.js

var averageOfLevels = function(root) {
    let ans = []
    let q = [];
    q.push(root);
    q.push(null);
    let sum=0.0;
    let cnt=0;
    while(q.length>0){
        let node = q[0];
        q.shift();
        if(node===null){
            ans.push(sum/cnt);
            sum=0.0;
            cnt=0;
            if(q.length > 0) q.push(null);
        }
        else{
            if(node.left) q.push(node.left);
            if(node.right) q.push(node.right);
            sum += node.val;
            cnt++;
        }
    }
    return ans;
};

File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.cpp

class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> t;
        queue<TreeNode*> q;
        q.push(root);
        q.push(nullptr);
        double sum=0.0;
        int cnt=0;
        while(!q.empty()){
            TreeNode* node = q.front();q.pop();
            if(node==nullptr){
                t.push_back(sum/cnt);
                sum=0.0;cnt=0;
                if(!q.empty())q.push(nullptr);
            }
            else{
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);
                sum += node->val;cnt++;
            }
            
        }
        return t;
    }
};